core/pkg/server/handler.go

```diff
@@ -215,8 +215,6 @@ func (h *Handler) handleRecord(record *service.Record) {
 		h.handleHeader(record)
 	case *service.Record_History:
 		h.handleHistory(x.History)
-	case *service.Record_LinkArtifact:
```

Comment:
keep this case to avoid panics. You can put it near the `Request_LinkArtifact` case and add a comment saying that it's an intentional no-op because it was removed.

Reply:
If we're keeping this case, shouldn't it just stay under the `handleRecord()` function?

Reply:
Also, I think `Record_LinkArtifact` will give a build error since we removed the record from the proto file, no?

Reply:
> If we're keeping this case, shouldn't it just stay under the handleRecord() function instead of handleRequest()?

Oh, yes good point. It should stay under `handleRecord()`; I missed that these were in different functions.

> Also, I think Record_LinkArtifact will give a build error since we removed the record from the proto file, no?

The name comes from the field name; I think it can be `Record_NoopLinkArtifact` now?

Reply:
Seems like the name comes from the first part - So that record type doesn't exist. I guess google.protobuf.Empty doesn't yield any new record type?

Reply:
The Record_NoopLinkArtifact type is generated here in your PR: https://github.com/wandb/wandb/blob/c8c5d3e631078072f9b802a4eb9f8b0e0f7ba2cc/core/pkg/service/wandb_internal.pb.go#L827-L830

Reply:
Oops, sorry. My IDE's go environment is probably messed up, so it didn't suggest this and I missed it. Added it now.